app/src/main/java/in/co/rajkumaar/amritarepo/activities/LaunchingActivity.java

```diff
@@ -577,8 +594,131 @@ private void intentSemActivity(int position, String title) {
         active = true;
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP){
+            // Create a listener to track request state updates.
+            listener = state -> {
+                // (Optional) Provide a download progress bar.
+                if (state.installStatus() == InstallStatus.DOWNLOADED) {
+                    popupSnackbarForCompleteUpdate();
+                }
+            };
+            // Before starting an update, register a listener for updates.
+            appUpdateManager.registerListener(listener);
+            appUpdateInfoTask.addOnSuccessListener(appUpdateInfo -> {
+                if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE){
+                    if(appUpdateInfo.updatePriority() > 3){
+                        try {
+                            appUpdateManager.startUpdateFlowForResult(
+                                    // Pass the intent that is returned by 'getAppUpdateInfo()'.
+                                    appUpdateInfo,
+                                    // Or 'AppUpdateType.FLEXIBLE' for flexible updates.
+                                    AppUpdateType.IMMEDIATE,
+                                    // The current activity making the update request.
+                                    this,
+                                    // Include a request code to later monitor this update request.
+                                    100);
+                        } catch (IntentSender.SendIntentException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                    else{
+                        try {
+                            appUpdateManager.startUpdateFlowForResult(
+                                    // Pass the intent that is returned by 'getAppUpdateInfo()'.
+                                    appUpdateInfo,
+                                    // Or 'AppUpdateType.FLEXIBLE' for flexible updates.
+                                    AppUpdateType.FLEXIBLE,
+                                    // The current activity making the update request.
+                                    this,
+                                    // Include a request code to later monitor this update request.
+                                    100);
+                        } catch (IntentSender.SendIntentException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                }
+            });
+        }
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == 100) {
+            if (resultCode == ActivityResult.RESULT_IN_APP_UPDATE_FAILED) {
+                Toast.makeText(getApplicationContext(),"The Update Failed, Starting again!",Toast.LENGTH_SHORT).show();
```

Comment:
Please don't hardcode strings. I know it exists in old codebase, but I'm trying to avoid them in future. And can the message be in sentence case ? (no uppercase letters in between)

Reply:
Also there's an utility function to create toasts. Please use them.